addons/Trusts/Trusts.lua

```diff
@@ -0,0 +1,338 @@
+        table.insert(party_ind,member.name)
+      end
+    end
+  end
+  return {party,party_ind};
+end
+
+function call_set(set)
+  queue = {} --trusts to be cast.
+  settings = config.load()
+  local party,party_ind = unpack(check_exist())
+  local limit = check_limit() --upper limit # of calls trust in current player.
+  local time = os.clock() --window open
+
+  if set == 'random' then
+    local checked = {}
+    local others = windower.ffxi.get_party().party1_count - #party_ind - 1 --# of human in party exept <me>.
+
+    if limit == #party_ind then
+      windower.send_command('input /retr all')
+      calls = limit
+      coroutine.sleep(settings.wait.retrall)
+    else
+      calls = limit - #party_ind - others
+    end
+
+    repeat
+      local index = trusts[math.random(1,#trusts)]
+      if not table.find(checked,index.name) then
+        table.insert(checked,index.name)
+        if windower.ffxi.get_spells()[index.id] and windower.ffxi.get_spell_recasts()[index.id] == 0 then
+          table.insert(queue,index)
+        end
+      end
+    until #queue >= calls or #checked >= 103 --# of unique names w/o Cornelia
+
+  elseif settings.sets[set] then
+    retr = {unpack(party_ind)}
+    for i=1,limit do
+      if settings.sets[set][tostring(i)] then
+        local entity = trusts:with(lang,settings.sets[set][tostring(i)])
+        if not party[entity.name]
+        or party[entity.name] ~= entity.models then
+          if windower.ffxi.get_spell_recasts()[entity.id] == 0 then
+            if windower.ffxi.get_spells()[entity.id] then
+              table.insert(queue,entity)
+            else
+              table.remove(retr,table.find(retr,party_ind[i]))
+              error('You aren\'t trusted by '..entity.english..'.')
+            end
+          else
+            table.remove(retr,table.find(retr,party_ind[i]))
+            local recast = math.floor(windower.ffxi.get_spell_recasts()[entity.id] / 6) / 10
+            log(entity.english..' needs '..recast..' secs break.')
+          end
+        else
+          table.remove(retr,table.find(retr,entity.name))
+          if settings.auto then
+            log(entity.english..' already exists.')
+          end
+        end
+      end
+    end
+    for index,name in pairs(retr) do
+      if #retr == #party_ind then
+        windower.send_command('input /retr all')
+        coroutine.sleep(settings.wait.retrall)
+        break;
+      else
+        windower.send_command('input /retr '..name)
+        coroutine.sleep(settings.wait.retr)
+      end
+    end
+  else
+      error('Unknown set name '..(set or ''))
+  end
+  --if /retr then wait at least 3secs.
+  local delay = (limit - #party_ind) == 0 and math.max(0,settings.wait.retrall + time - os.clock()) or 0
+  coroutine.schedule(call_trust,delay)
+end
+
+windower.register_event('action', function(act)
+  if settings.auto and act.actor_id == windower.ffxi.get_player().id and queue and #queue > 0 then
```

Comment:
You should probably have a global `player` variable and only set it on the `login` event to avoid calling this function over and over.

Reply:
done. calls with when `login` and `load` event is fired.